pkg/controller/clusterpool/clusterpool_controller.go

```diff
@@ -386,6 +401,15 @@ func (r *ReconcileClusterPool) Reconcile(ctx context.Context, request reconcile.
 	return reconcile.Result{}, nil
 }
 
+func calculatePoolVersion(clp *hivev1.ClusterPool) string {
+	ba := []byte{}
```

Comment:
since we store the hashes in the annotation, what is the expected max length of the hash? the append here makes it seems like it grows with more fields we include?

Reply:
> Fortunately for us, there's effectively no limit to the length of an annotation value :)

even though there is no limitation, i would expect that our hash is capped/constant at a length like most hashes.. our hashed value that keep increasing in length just looks odd and might create confusion in future imo.

Reply:
Changed to hash the hashes, which fixes the length at 16c.

(ed. Notwithstanding what I said above, anyone confused about the actual value of the string shouldn't be looking at the string. It's supposed to be an opaque value -- again like kube's `resourceVersion`.)